### PR TITLE
@ashfurrow => Migrate to AFNetworking 2.0

### PR DESCRIPTION
**Pull request by orta**
Things to do:

- [x] Tests to pass
- [ ] Auth to be verified manually
- [x] Grouped networking requests previously using `enqueueBatchOfHTTPRequestOperations` needs to be replaced with `dispatch_queues`?
- [x] bring back reachability

### CODE REVIEW

**Comment by orta** [comment]
screen shot 2015-07-21 at 15 10 16

Yeah, netwooorrkkking is working again.

**Comment by orta** [comment]
OK, this is ready for review. I'll go through and see what could need additional annotation.

File: Artsy/Networking/API_Modules/ArtsyAPI+Artworks.m [hunk 1]
```diff
@@ -44,64 +44,70 @@ + (AFJSONRequestOperation *)createPendingOrderWithArtworkID:(NSString *)artworkI
...
+    NSURLRequest *bidderPositionRequest = [ARRouter bidderPositionsRequestForSaleID:saleID artworkID:artworkID];
 
-    // get bidder position
-    request = [ARRouter bidderPositionsRequestForSaleID:saleID artworkID:artworkID];
-    AFJSONRequestOperation *positionsOperation = [AFJSONRequestOperation JSONRequestOperationWithRequest:request success:nil failure:nil];
-
-    NSArray *operations = @[ biddersOperation, saleArtworkOperation, positionsOperation ];
-    AFHTTPClient *client = [ARRouter httpClient];
-    [client enqueueBatchOfHTTPRequestOperations:operations progressBlock:nil completionBlock:^(NSArray *operations) {
+    [self getRequests:@[saleArtworkRequest, biddersRequest, bidderPositionRequest] success:^(NSArray *operations) {
```
Comment (orta) [inline Artsy/Networking/API_Modules/ArtsyAPI+Artworks.m]: now the queuing is done at the private `ArtsyAPI` level

File: Artsy/Networking/ARAuthProviders.m [hunk 2]
```diff
@@ -19,14 +19,15 @@ + (void)getReverseAuthTokenForTwitter:(void (^)(NSString *token, NSString *secre
...
                                               secret:[ArtsyKeys new].artsyTwitterSecret];
 
-    [client authorizeUsingOAuthWithRequestTokenPath:@"/oauth/request_token"
-        userAuthorizationPath:@"/oauth/authorize"
+
+    [client authorizeUsingOAuthWithRequestTokenURLString:@"/oauth/request_token"
+        userAuthorizationURLString:@"/oauth/authorize"
         callbackURL:[NSURL URLWithString:ARTwitterCallbackPath]
-        accessTokenPath:@"/oauth/access_token"
+        accessTokenURLString:@"/oauth/access_token"
```
Comment (orta) [inline Artsy/Networking/ARAuthProviders.m]: path -> URLString is one of the biggies in AFNetworking 1 -> 2

File: Artsy/Networking/ARNetworkConstants.m [hunk 3]
```diff
@@ -1,10 +1,10 @@
-NSString *const ARBaseDesktopWebURL = @"https://artsy.net/";
-NSString *const ARBaseMobileWebURL = @"https://m.artsy.net/";
-NSString *const ARBaseApiURL = @"https://api.artsy.net/";
+NSString *const ARBaseDesktopWebURL = @"https://artsy.net";
```
Comment (orta) [inline Artsy/Networking/ARNetworkConstants.m]: the AF2 HTTP client-like manager appends its own `/` now

File: Artsy/Networking/ARRouter.m [hunk 4]
```diff
@@ -675,41 +703,41 @@ + (NSURLRequest *)directImageRequestForModel:(Class)model andSlug:(NSString *)sl
...
     NSString *url = [NSString stringWithFormat:ARNewFairShowsURLFormat, fair.fairID];
-    return [staticHTTPClient requestWithMethod:@"GET" path:url parameters:nil];
+    return [self requestWithMethod:@"GET" path:url parameters:nil];
 }
 
 + (NSURLRequest *)newFairMapRequestWithFair:(Fair *)fair
 {
-    NSString *url = [NSString stringWithFormat:ARNewFairMapURLFormat, fair.fairID];
-    return [staticHTTPClient requestWithMethod:@"GET" path:url parameters:nil];
+    NSString *url = [NSString stringWithFormat:ARNewFairMapURL];
```
Comment (orta) [inline Artsy/Networking/ARRouter.m]: this URL string had params built into it

File: Artsy/Networking/ArtsyAPI.m [hunk 5]
```diff
@@ -1,44 +1,61 @@
...
 }
 
++ (void)getRequests:(NSArray *)requests success:(void (^)(NSArray *operations))completed
+{
+    [self.sharedAPI getRequests:requests success:completed];
+}
+
+- (void)getRequests:(NSArray *)requests success:(void (^)(NSArray *operations))completed
+{
+    NSArray *operations = [requests map:^id(NSURLRequest *request) {
```
Comment (ashfurrow) [inline Artsy/Networking/ArtsyAPI.m]: Very nice :clap:

File: Artsy/Networking/ArtsyAPI.m [hunk 6]
```diff
@@ -211,31 +232,33 @@ - (AFJSONRequestOperation *)getRequest:(NSURLRequest *)request parseIntoAnArrayO
...
-        }
-        }];
-    getOperation.successCallbackQueue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+        ar_dispatch_main_queue(^{
+            if (failure) {
+                failure(error);
+            }
+        });
+    }];
+    getOperation.completionQueue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
```
Comment (ashfurrow) [inline Artsy/Networking/ArtsyAPI.m]: To be clear, this is not the main queue, right? So the success/failure blocks won't be updating UI? Or they will through dispatching?

Comment (orta) [inline Artsy/Networking/ArtsyAPI.m]: They all do their parsing on a BG thread, then the success blocks go back to the main

File: Artsy_Tests/View_Controller_Tests/Web_Browsing/ARInternalMobileWebViewControllerTests.m [hunk 7]
```diff
@@ -191,10 +192,15 @@ @interface ARInternalMobileWebViewController (Testing)
         beforeEach(^{
             controller = [[ARInternalMobileWebViewController alloc] initWithURL:[NSURL URLWithString:@""]];
         });
-        
-        it(@"handles an internal link being clicked", ^{
+
+        it(@"handles a non-native internal link being clicked", ^{
+            NSURLRequest *request = [controller requestWithURL:[NSURL URLWithString:@"/something/andy-warhol-skull"]];
+            expect([controller webView:nil shouldStartLoadWithRequest:request navigationType:UIWebViewNavigationTypeLinkClicked]).to.beFalsy();
```
Comment (ashfurrow) [inline Artsy_Tests/View_Controller_Tests/Web_Browsing/ARInternalMobileWebViewControllerTests.m]: In the test below, you pass in `controller.webView` – is there a significance to this difference?

Comment (orta) [inline Artsy_Tests/View_Controller_Tests/Web_Browsing/ARInternalMobileWebViewControllerTests.m]: yes, its a new test and therefore didn't include stuff from @alloy  - will add.

File: Artsy/Networking/ArtsyAPI.m (hunk 6 above)
Comment (ashfurrow) [inline Artsy/Networking/ArtsyAPI.m]: OK just checking ;+1:

**Pull request merged**